src/engine.ts

```diff
@@ -55,20 +56,45 @@ export default class LuaEngine {
         }
     }
 
+    /**
+     * Executes Lua code from a string asynchronously.
+     * @param script - Lua script to execute.
+     * @returns A Promise that resolves to the result returned by the Lua script execution.
+     * @throws {Error}
+     */
     public doString(script: string): Promise<any> {
         return this.callByteCode((thread) => thread.loadString(script))
     }
 
+    /**
+     * Executes Lua code from a file asynchronously.
+     * @async
+     * @param filename - Path to the Lua script file.
+     * @returns - A Promise that resolves to the result returned by the Lua script execution.
+     * @throws {Error}
+     */
     public doFile(filename: string): Promise<any> {
         return this.callByteCode((thread) => thread.loadFile(filename))
     }
 
+    /**
+     * Executes Lua code from a string synchronously.
+     * @param script - Lua script to execute.
+     * @returns - The result returned by the Lua script.
+     * @throws {Error}
+     */
     public doStringSync(script: string): any {
         this.global.loadString(script)
         const result = this.global.runSync()
         return result[0]
     }
 
+    /**
+     * Executes Lua code from a file synchronously.
+     * @param filename - Path to the Lua script file.
+     * @returns - The result returned by the Lua script.
+     * @throws {Error}
```

Comment:
@throws with the default Error object is kinda useless?

Reply:
Some libraries throw different errors, tought it could add to something, but it can be removed just fine